packages/log-view/src/browser/log-view-contribution.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { MenuContribution, CommandContribution, MenuModelRegistry, CommandRegistry, Command } from "@theia/core";
+import { WidgetManager, FrontendApplication, CommonMenus } from "@theia/core/lib/browser";
+import { inject, injectable, multiInject } from "inversify";
+import { LOG_VIEW_KIND } from "./log-view-widget";
+import { LanguageClientContribution } from "@theia/languages/lib/browser";
+import { LogModel } from "./log-model";
+import { LogMessageNotification } from "@theia/languages/lib/common";
+
+export namespace LogViewCommands {
+    export const OPEN: Command = {
+        id: 'logView:open',
+        label: 'Open Log View'
+    };
+}
+
+@injectable()
+export class LogViewContribution implements CommandContribution, MenuContribution {
+
+    constructor(
+        @inject(WidgetManager) protected readonly widgetManager: WidgetManager,
+        @inject(FrontendApplication) protected readonly app: FrontendApplication,
+        @multiInject(LanguageClientContribution) languageClients: LanguageClientContribution[],
+        @inject(LogModel) logModel: LogModel) {
+        languageClients.forEach(contrib => {
+            contrib.languageClient.then(
+                languageClient => {
+                    const logChannel = logModel.getChannel(contrib.name);
+                    languageClient.onNotification(LogMessageNotification.type, logChannel.log.bind(logChannel));
+                });
+        });
+    }
+
+    registerCommands(commands: CommandRegistry): void {
+        commands.registerCommand(LogViewCommands.OPEN, {
+            execute: () => this.openLogView()
+        });
+    }
+
+    registerMenus(menus: MenuModelRegistry): void {
+        menus.registerMenuAction(CommonMenus.VIEW, {
+            commandId: LogViewCommands.OPEN.id
+        });
+    }
+
+    protected async openLogView(): Promise<void> {
```

Comment:
would be nice if it is public and returns an instance of `LogWidget`

Reply:
Done